src/posix_regex.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_posix_regex_h__
+#define INCLUDE_posix_regex_h__
+
+#include <regex.h>
+
+/*
+ * Regular expressions: if the operating system has p_regcomp_l,
+ * use that as our p_regcomp implementation, otherwise fall back
+ * to standard regcomp.
+ */
+
+#define P_REG_EXTENDED REG_EXTENDED
+#define P_REG_ICASE REG_ICASE
+#define P_REG_NOMATCH REG_NOMATCH
+
+#define p_regex_t regex_t
+#define p_regmatch_t regmatch_t
+
+#define p_regerror regerror
+#define p_regexec regexec
+#define p_regfree regfree
+
+#ifdef GIT_USE_REGCOMP_L
```

Comment:
This is broken, right? If you directly included "posix_regex.h", this define might never be set due to the header missing an include of "common.h"->"features.h"

Reply:
This has been fixed :+1: